airflow/providers/microsoft/azure/transfers/azure_blob_to_gcs.py

```diff
@@ -17,102 +17,26 @@
 # under the License.
 from __future__ import annotations
 
-import tempfile
-from typing import TYPE_CHECKING, Sequence
+import warnings
 
-from airflow.models import BaseOperator
-from airflow.providers.google.cloud.hooks.gcs import GCSHook
-from airflow.providers.microsoft.azure.hooks.wasb import WasbHook
+from airflow.exceptions import AirflowProviderDeprecationWarning
+from airflow.providers.google.cloud.transfers.azure_blob_to_gcs import (
+    AzureBlobStorageToGCSOperator as AzureBlobStorageToGCSOperatorFromGoogleProvider,
+)
 
-if TYPE_CHECKING:
-    from airflow.utils.context import Context
 
-
-class AzureBlobStorageToGCSOperator(BaseOperator):
+class AzureBlobStorageToGCSOperator(AzureBlobStorageToGCSOperatorFromGoogleProvider):
     """
-    Operator transfers data from Azure Blob Storage to specified bucket in Google Cloud Storage.
-
-    .. seealso::
-        For more information on how to use this operator, take a look at the guide:
-        :ref:`howto/operator:AzureBlobStorageToGCSOperator`
-
-    :param wasb_conn_id: Reference to the wasb connection.
-    :param gcp_conn_id: The connection ID to use when fetching connection info.
-    :param blob_name: Name of the blob
-    :param container_name: Name of the container
-    :param bucket_name: The bucket to upload to
-    :param object_name: The object name to set when uploading the file
-    :param filename: The local file path to the file to be uploaded
-    :param gzip: Option to compress local file or file data for upload
-    :param impersonation_chain: Optional service account to impersonate using short-term
-        credentials, or chained list of accounts required to get the access_token
-        of the last account in the list, which will be impersonated in the request.
-        If set as a string, the account must grant the originating account
-        the Service Account Token Creator IAM role.
-        If set as a sequence, the identities from the list must grant
-        Service Account Token Creator IAM role to the directly preceding identity, with first
-        account from the list granting this role to the originating account.
+    This class is deprecated.
+    Please use `airflow.providers.google.cloud.transfers.azure_blob_to_gcs.AzureBlobStorageToGCSOperator`.
```

Comment:
I've been working on enabling rule D205 in the stylechecks which enforces a blank line after the summary, if you don't mind.

```suggestion
    This class is deprecated.
    
    Please use `airflow.providers.google.cloud.transfers.azure_blob_to_gcs.AzureBlobStorageToGCSOperator`.
```